rkrishn7/couchsync
# added host page transition changes/notifications

### 📃 Summary
Added code to the event listeners page to dispatch URL Changes and emit the events to the socket only if user is host. My spaghetti code is pretty minimal, but may need some prettifying or changes.

### 🔍 What's the new behavior?
Now the party URL will update along with a change in videos on in-page navigation, along with sending a socket event to the server.

### 🏷 Task Link
[Story link on Notiono](https://www.notion.so/Handle-Page-Transition-Events-f4a65c8f34f9419aae500dea7c800fd0)

### ↩️ Depends On

Link any other PRs here.

### 🧪 QA

- List the steps to test the changes here.
- Run the extension
- Create a party 
- Navigate from one video to another, check that URL in party popup changes accordingly

### 🛑 Risk Analysis

- [ ] Does it update dependencies?
